src/app/store/room/room.reducer.ts

```diff
@@ -63,6 +64,62 @@ const _roomReducer = createReducer<RoomState>(initialRoomState,
     error,
     status: RoomStatus.error
   })),
+  on(RoomAction.selectCard, state => ({
+    ...state,
+    status: RoomStatus.loading
+  })),
+  on(RoomAction.cardSelectionSuccess, (state, {vote}) => {
+    let map: Map<Participant, Card> = new Map<Participant, Card>([]);
+    if (state.room.votingResult.map instanceof Map) {
+      map = new Map<Participant, Card>(state.room.votingResult.map);
+    }
+    let found: boolean = false;
+    for (let entry of map.entries()) {
+      if (entry[0].nickname == vote.participant.nickname) {
+        map.set(entry[0], vote.card);
+      }
+    }
+    if (!found) {
+      map.set(vote.participant, vote.card);
+    }
+    return {
+      ...state,
+      room: {
+        ...state.room,
+        votingResult: {
+          ...state.room.votingResult,
+          map
+        }
+      },
+      status: RoomStatus.success
+    };
+  }),
```

Comment:
This reducer is really big. This is due to the fact that I need to copy the map and add a new value. Do you know how I can improve it?

Reply:
I can use cloneDeep() from lodash